<a href="https://colab.research.google.com/github/bam396/nyu-python-public-policy/blob/master/HW2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Homework #2**
**Brittany Mazzurco Muscato **

In [4]:

import pandas as pd 
from google.colab import drive 
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [5]:
df = pd.read_csv('/content/drive/My Drive/Data for Python/cleaned_311_data.csv', header='infer')

/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (9,18,21,32,35) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [6]:
df.head()

Unnamed: 0  unique_key  ...  longitude                                 location
0           0    39888071  ... -73.942942   (40.79333937834769, -73.9429417746998)
1           1    39889166  ... -73.953071   (40.82012422332215, -73.9530712339799)
2           2    39882869  ... -73.951399   (40.67080917938279, -73.9513990916184)
3           3    39894246  ... -73.990517  (40.74026158873342, -73.99051651686905)
4           4    39881329  ... -73.750990  (40.68814402968042, -73.75098958473612)

[5 rows x 42 columns]

# Question 1

In [17]:
counts = df.groupby('open_data_channel_type').size().nlargest(15).reset_index(name='count')
counts

open_data_channel_type    count
0                  PHONE  1463204
1                 ONLINE   589865
2                UNKNOWN   417916
3                 MOBILE   368759
4                  OTHER    19267

The most common complaint submission method is by phone with 1,463,204 complaints.

In [11]:
df.columns

Index(['Unnamed: 0', 'unique_key', 'created_date', 'closed_date', 'agency',
       'agency_name', 'complaint_type', 'descriptor', 'location_type',
       'incident_zip', 'incident_address', 'street_name', 'cross_street_1',
       'cross_street_2', 'intersection_street_1', 'intersection_street_2',
       'address_type', 'city', 'landmark', 'facility_type', 'status',
       'due_date', 'resolution_description', 'resolution_action_updated_date',
       'community_board', 'bbl', 'borough', 'x_coordinate_(state_plane)',
       'y_coordinate_(state_plane)', 'open_data_channel_type',
       'park_facility_name', 'park_borough', 'vehicle_type',
       'taxi_company_borough', 'taxi_pick_up_location', 'bridge_highway_name',
       'bridge_highway_direction', 'road_ramp', 'bridge_highway_segment',
       'latitude', 'longitude', 'location'],
      dtype='object')

# Question 2

In [16]:
counts_type = df.groupby(['open_data_channel_type', 'complaint_type']).size().nlargest(15).reset_index(name='count')
counts_type

open_data_channel_type                       complaint_type   count
0                   PHONE  Request Large Bulky Item Collection  177175
1                   PHONE                       HEAT/HOT WATER  128002
2                   PHONE                  Noise - Residential   95407
3                   PHONE                     Blocked Driveway   95206
4                  ONLINE                  Noise - Residential   89050
5                 UNKNOWN               Street Light Condition   77755
6                  MOBILE                      Illegal Parking   75784
7                   PHONE                      Illegal Parking   63957
8                   PHONE                 UNSANITARY CONDITION   63619
9                 UNKNOWN                     Street Condition   61508
10                 ONLINE                      Illegal Parking   55417
11                UNKNOWN        General Construction/Plumbing   54939
12                 ONLINE                       HEAT/HOT WATER   54872
13                 MOBILE                  Noise - Residential   51889
14                  PHONE                        PAINT/PLASTER   46163

The most common complaint was to request large bulky item collection, which was made via phone 177,175 times. In my opinion, the reason why most requests are made via phone or mobile is because the name of this service/department ("311") is a phone code, so its name alone suggests contact by phone. The few complaint types made by Desktop computer ("Noise - Residential" and "HEAT/HOT WATER") were ilkely made by individuals already home when they experienced this problem, and they were likely looking for the service/department to call to fix this issue.

# Question 3

In [28]:
complaint_counts = df.groupby(['complaint_type', 'open_data_channel_type']).size().reset_index(name='count')

complaint_counts

complaint_type open_data_channel_type  count
0                  APPLIANCE                 ONLINE   2627
1                  APPLIANCE                  PHONE  12002
2          Abandoned Vehicle                 MOBILE   2484
3          Abandoned Vehicle                 ONLINE   1891
4          Abandoned Vehicle                  PHONE   5069
..                       ...                    ...    ...
514             Window Guard                 ONLINE     17
515             Window Guard                  PHONE     13
516             Window Guard                UNKNOWN      1
517  X-Ray Machine/Equipment                 ONLINE      5
518  X-Ray Machine/Equipment                  PHONE      4

[519 rows x 3 columns]

In [30]:
complaint_counts.sort_values(['count','complaint_type'], ascending=[False,True]).drop_duplicates(['complaint_type'])

complaint_type open_data_channel_type   count
378  Request Large Bulky Item Collection                  PHONE  177175
200                       HEAT/HOT WATER                  PHONE  128002
316                  Noise - Residential                  PHONE   95407
47                      Blocked Driveway                  PHONE   95206
429               Street Light Condition                UNKNOWN   77755
..                                   ...                    ...     ...
52                         Bottled Water                  PHONE       5
165            Fire Alarm - Reinspection                 ONLINE       5
182                 Forensic Engineering                UNKNOWN       5
374                            Rangehood                 ONLINE       5
517              X-Ray Machine/Equipment                 ONLINE       5

[229 rows x 3 columns]

Double-checking how many unique complaint types there are (229).

In [32]:
df.groupby('complaint_type').size().reset_index(name='count')

complaint_type  count
0                  APPLIANCE  14629
1          Abandoned Vehicle   9444
2            Advocate - Lien      9
3           Advocate - Other    129
4            Advocate - RPIE      5
..                       ...    ...
224       Water Conservation   4641
225            Water Quality   1910
226             Water System  74139
227             Window Guard     31
228  X-Ray Machine/Equipment      9

[229 rows x 2 columns]

# **The End!**